### PR TITLE
Add pre-reqs to VMSS encryption templates

### PR DESCRIPTION
**Pull request by SudhakaraReddyEvuri**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by SudhakaraReddyEvuri** [comment]
@bmoore-msft  can you please review this PR to add pre-reqs for VMSS encryption templates

File: 201-encrypt-running-vmss-windows/azuredeploy.json [hunk 1]
```diff
@@ -44,51 +44,43 @@
...
-            ]
-          }
+        "publisher": "Microsoft.Azure.Security",
+        "type": "AzureDiskEncryption",
+        "typeHandlerVersion": "2.2",
+        "autoUpgradeMinorVersion": true,
+        "forceUpdateTag": "[parameters('forceUpdateTag')]",
+        "settings": {
+          "EncryptionOperation": "EnableEncryption",
+          "KeyVaultURL": "[reference(variables('keyVaultResourceId'),'2016-10-01').vaultUri]",
```
Comment (bmoore-msft) [inline 201-encrypt-running-vmss-windows/azuredeploy.json]: updated apiVersion

File: 201-decrypt-vmss-windows/README.md [hunk 2]
```diff
@@ -7,4 +7,12 @@
...
 
-This template disables encryption on the VM Scale Set of Windows VMs.
+This template disables encryption on a running VM Scale Set of Windows VMs.
+
+__Note: The VMSS encryption preview does not yet support image upgrade or reimage. Do not use this if you will need to upgrade your OS image in an encrypted scale set.__
+
+AzureDiskEncryption for VMSS is currently in preview. Consuming this feature requires enabling the preview feature on the subscription and setting up a key vault with 'EnabledForDiskEncryption' access policy using the Azure powershell cmdlets below 
+1. Set-AzureRmKeyVaultAccessPolicy -ResourceGroupName <rgName> -VaultName <vaultName> -EnabledForDiskEncryption"
+2. Register-AzureRmProviderFeature -FeatureName "UnifiedDiskEncryption" -ProviderNamespace "Microsoft.Compute"
+3. Wait 10 min till state transitions to 'Registered'
```
Comment (bmoore-msft) [inline 201-decrypt-vmss-windows/README.md]: Wait 10 minutes until ...

File: 201-decrypt-vmss-windows/prereqs/prereq.azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        "loadBalancerName": "[concat(variables('namingInfix'), 'lb')]",
+        "natPoolName": "[concat(variables('namingInfix'), 'natpool')]",
+        "bePoolName": "[concat(variables('namingInfix'), 'bepool')]",
+        "keyVaultResourceID": "[resourceId(parameters('keyVaultResourceGroup'), 'Microsoft.KeyVault/vaults/', parameters('keyVaultName'))]"
+    },
+    "resources": [{
+            "type": "Microsoft.Network/virtualNetworks",
+            "name": "[variables('virtualNetworkName')]",
+            "location": "[resourceGroup().location]",
+            "apiVersion": "2016-03-30",
```
Comment (bmoore-msft) [inline 201-decrypt-vmss-windows/prereqs/prereq.azuredeploy.json]: update apiVersions throughout...

File: 201-encrypt-running-vmss-windows/README.md [hunk 4]
```diff
@@ -12,6 +12,7 @@ This template enables encryption on a running VM Scale Set of Windows VMs.
 __Note: The VMSS encryption preview does not yet support image upgrade or reimage. Do not use this if you will need to upgrade your OS image in an encrypted scale set.__
 
 AzureDiskEncryption for VMSS is currently in preview. Consuming this feature requires enabling the preview feature on the subscription and setting up a key vault with 'EnabledForDiskEncryption' access policy using the Azure powershell cmdlets below 
-1. Register-AzureRmProviderFeature -FeatureName "UnifiedDiskEncryption" -ProviderNamespace "Microsoft.Compute"
-2. Set-AzureRmKeyVaultAccessPolicy -ResourceGroupName <rgName> -VaultName <vaultName> -EnabledForDiskEncryption"
-
+1. Set-AzureRmKeyVaultAccessPolicy -ResourceGroupName <rgName> -VaultName <vaultName> -EnabledForDiskEncryption"
+2. Register-AzureRmProviderFeature -FeatureName "UnifiedDiskEncryption" -ProviderNamespace "Microsoft.Compute"
+3. Wait 10 min till state transitions to 'Registered'
```
Comment (bmoore-msft) [inline 201-encrypt-running-vmss-windows/README.md]: Wait 10 minutes until ...

File: 201-encrypt-running-vmss-windows/prereqs/prereq.azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+        "subnetName": "[concat(variables('namingInfix'), 'subnet')]",
+        "loadBalancerName": "[concat(variables('namingInfix'), 'lb')]",
+        "natPoolName": "[concat(variables('namingInfix'), 'natpool')]",
+        "bePoolName": "[concat(variables('namingInfix'), 'bepool')]"
+    },
+    "resources": [{
+            "type": "Microsoft.Network/virtualNetworks",
+            "name": "[variables('virtualNetworkName')]",
+            "location": "[resourceGroup().location]",
+            "apiVersion": "2016-03-30",
```
Comment (bmoore-msft) [inline 201-encrypt-running-vmss-windows/prereqs/prereq.azuredeploy.json]: update apiVersions throughout...

**Comment by SudhakaraReddyEvuri** [comment]
@bmoore-msft  addressed your comments. Please review the PR.

**Pull request merged**